components/cop_codegen/src/rpn_function.rs

```diff
@@ -573,6 +745,26 @@ impl NormalRpnFn {
             evaluator = quote! { ArgConstructor::new(#arg_index, #evaluator) };
         }
         let fn_name = self.item_fn.ident.to_string();
+
+        let validator_fn;
+        let validator_fn_ptr;
+        if self.validator.is_none() {
+            validator_fn = Some(
+                ValidatorFnGenerator::new()
+                    .validate_return_type(&self.ret_type)
+                    .validate_args_type(&self.arg_types)
+                    .generate(&impl_generics, where_clause),
+            );
+            validator_fn_ptr = quote! { validate #ty_generics_turbofish };
+        } else {
+            validator_fn = None;
+            validator_fn_ptr = self
+                .validator
+                .as_ref()
+                .map(|f| f.into_token_stream())
+                .unwrap();
+        }
```

Comment:
This part of code looks similar among the three types of RPN functions. Can we extract it out?